### PR TITLE
stubbed out beehave_tree_node

### PR DESCRIPTION
**Pull request by ryan-linehan**
## Description

Stubbed out methods from beehave 2.X's behave node (in beehave 3.X it is renamed to beehave_tree_node). Implemented the methods I could without the blackboard being completely implemented.

## Screenshots
configuration warnings working with this update:
image

docs updated with new methods:
image

### CODE REVIEW

File: extension/src/nodes/beehave_tree_node.cpp [hunk 1]
```diff
@@ -38,14 +38,59 @@ BeehaveTreeNode::BeehaveTreeNode() {
...
+			break;
+		}
+	}
+
+	return warnings;
+}
+
 BeehaveTreeNode::TickStatus BeehaveTreeNode::tick(Ref<BeehaveContext> context) {
-	return BeehaveTreeNode::FAILURE;
+	return BeehaveTreeNode::SUCCESS;
```
Comment (ryan-linehan) [inline extension/src/nodes/beehave_tree_node.cpp:59]: Changed this to success as this is what the 2.X version was doing when I looked. I can swap it back if we want

File: extension/src/nodes/beehave_tree_node.cpp [hunk 2]
```diff
@@ -38,14 +38,59 @@ BeehaveTreeNode::BeehaveTreeNode() {
...
 
 void BeehaveTreeNode::_bind_methods() {
+
 	ClassDB::bind_method(D_METHOD("tick"), &BeehaveTreeNode::tick);
+	ClassDB::bind_method(D_METHOD("interrupt"), &BeehaveTreeNode::interrupt);
+	ClassDB::bind_method(D_METHOD("before_run"), &BeehaveTreeNode::before_run);
+	ClassDB::bind_method(D_METHOD("after_run"), &BeehaveTreeNode::after_run);
+	ClassDB::bind_method(D_METHOD("get_class_name"), &BeehaveTreeNode::get_class_name);
+	ClassDB::bind_method(D_METHOD("can_send_message"), &BeehaveTreeNode::can_send_message);
+	/*ClassDB::bind_method(D_METHOD("get_configuration_warnings"), &BeehaveTreeNode::get_configuration_warnings);*/
```
Comment (ryan-linehan) [inline extension/src/nodes/beehave_tree_node.cpp]: will want to remove this comment. I didn't realize that get configuration warnings was a method we would be overriding

File: extension/src/nodes/beehave_tree_node.cpp [hunk 3]
```diff
@@ -38,14 +38,58 @@ BeehaveTreeNode::BeehaveTreeNode() {
...
 }
 
 void BeehaveTreeNode::_bind_methods() {
+
 	ClassDB::bind_method(D_METHOD("tick"), &BeehaveTreeNode::tick);
+	ClassDB::bind_method(D_METHOD("interrupt"), &BeehaveTreeNode::interrupt);
+	ClassDB::bind_method(D_METHOD("before_run"), &BeehaveTreeNode::before_run);
+	ClassDB::bind_method(D_METHOD("after_run"), &BeehaveTreeNode::after_run);
+	ClassDB::bind_method(D_METHOD("get_class_name"), &BeehaveTreeNode::get_class_name);
+	ClassDB::bind_method(D_METHOD("can_send_message"), &BeehaveTreeNode::can_send_message);
```
Comment (bitbrain) [inline extension/src/nodes/beehave_tree_node.cpp:69]:
```
ClassDB::bind_method(D_METHOD("get_class_name"), &BeehaveTreeNode::get_class_name);
ClassDB::bind_method(D_METHOD("can_send_message"), &BeehaveTreeNode::can_send_message);
```
Those were needed for the debugger. Let us (for now) not define those methods and deal with that later, as with C++ we can do some fancy tricks to not rely on GDScript hacks.

File: extension/src/nodes/beehave_tree_node.cpp [hunk 4]
```diff
@@ -38,14 +38,58 @@ BeehaveTreeNode::BeehaveTreeNode() {
...
 BeehaveTreeNode::TickStatus BeehaveTreeNode::tick(Ref<BeehaveContext> context) {
-	return BeehaveTreeNode::FAILURE;
+	return BeehaveTreeNode::SUCCESS;
 }
 
 void BeehaveTreeNode::_bind_methods() {
+
 	ClassDB::bind_method(D_METHOD("tick"), &BeehaveTreeNode::tick);
+	ClassDB::bind_method(D_METHOD("interrupt"), &BeehaveTreeNode::interrupt);
+	ClassDB::bind_method(D_METHOD("before_run"), &BeehaveTreeNode::before_run);
```
Comment (bitbrain) [inline extension/src/nodes/beehave_tree_node.cpp:66]: I was thinking about making `before_run` and `after_run` signals in V3

**Comment by bitbrain** [comment]
Closing this, as this has been done already.

**Pull request closed without merging**